What's New 100 (#1817)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/whats-new/2022/03/devtools-100.md

If you are on Windows, Linux, or macOS, consider using the Canary preview channel of Microsoft Edge as your default development browser.  The preview channels give you access to the latest features of Microsoft Edge DevTools.

To report a problem with DevTools or ask for a new feature, file an issue at the new [Microsoft Edge DevTools feedback repo](https://github.com/MicrosoftEdge/DevTools).

<!-- ====================================================================== -->
## Join the Microsoft Edge DevTools community on GitHub

<!-- Title: Head to the new DevTools repo on GitHub to send ideas, feedback, suggestions, and bugs -->
<!-- Subtitle: You can file feedback, ask questions, and have discussions about DevTools in our GitHub repo. -->

> We're excited to announce the new [MicrosoftEdge/DevTools: Feedback and discussions about Microsoft Edge Developer Tools (github.com)](https://github.com/MicrosoftEdge/DevTools) repo on GitHub!  This new repo is a place for our community of developers to send us ideas, feedback, suggestions, and tips to improve DevTools. We're eager to discuss with you how DevTools can better serve your needs and meet the challenges you face every day in building great web experiences.

Reviewer: I feel like we're including the default link text that Edge provides when you copy and paste a link and I think it's unnecessarily verbose. Here's my suggestion:
```suggestion
We're excited to announce the new [MicrosoftEdge/DevTools](https://github.com/MicrosoftEdge/DevTools) repo on GitHub!  This new repo is a place for our community of developers to send us ideas, feedback, suggestions, and tips to improve DevTools. We're eager to discuss with you how DevTools can better serve your needs and meet the challenges you face every day in building great web experiences.
```

![The DevTools feedback repo.](devtools-100-images/devtools-feedback-repo.png)

Head to [our repo on GitHub](https://github.com/MicrosoftEdge/DevTools) to become a part of the Edge DevTools community.

<!-- ====================================================================== -->
## Filter heap snapshots summary by node type

<!-- Title: Use new filters to focus on specific parts of a heap snapshot -->
<!-- Subtitle: You can now filter by node type if, for example, you're only interested in the arrays or strings from the heap. -->

When looking at all the objects in a heap snapshot in the **Memory** tool, it can be difficult to focus on specific objects or retaining paths.  Now in Microsoft Edge 100, you can use the new **Node Types** filter when looking at a heap snapshot, to focus on only specific types of nodes.  For example, to see only the arrays and string objects that are in the heap, select the **Array** and **String** entries in the **Node Types** filter.

![Node Types in a heap snapshot in the Memory tool.](devtools-100-images/node-types-heap-snapshot.png)

> To learn more about taking a heap snapshot and analyzing the heap in DevTools, see [Record heap snapshots using the Memory tool](../../../memory-problems/heap-snapshots.md).

Reviewer: Thank you for adding a new section to the linked doc. The additional section looks great to me.

<!-- ====================================================================== -->
## Use DevTools in Czech and Vietnamese

<!-- Title: DevTools: Now available in Czech and Vietnamese -->
<!-- Subtitle: Try out DevTools in your preferred language!  If we haven't supported it, yet, let us know. -->

Customizing and simplifying the developer experience includes making our tools available to everyone, regardless of their preferred language.  We've added two new languages, Czech and Vietnamese, to our list of supported languages, and will keep adding more.  If your preferred language isn't supported in DevTools, let us know!

![DevTools in Czech and Vietnamese.](devtools-100-images/czech-vietnamese.png)

> For more information, see [Change DevTools language settings](../../../customize/localization.md).

Reviewer: Just like you did for heap-snapshots.md, please Czech and Vietnamese to the list of supported languages in customize/localization.md. Follow up with Erica if you have any questions!

<!-- ====================================================================== -->
## The Network tool now displays how a request was fulfilled

<!-- Title: You no longer have to wonder if a request was fulfilled by your service worker or cache -->
<!-- Subtitle: The "Fulfilled by" column in the Network tool tells you how a request was fulfilled. -->

> In Microsoft Edge 100, the **Network** tool supports a new, **Fulfilled by** column in the request log.  The **Fulfilled by** column shows how a request was fulfilled.  If the request was fulfilled by your service worker or from the cache, that information is now tracked in the **Fulfilled by** column.  When exporting or importing the request log as a HAR file, the information in the **Fulfilled by** column is also included.

Reviewer 2: ```suggestion
In Microsoft Edge 100, the **Network** tool supports a new, **Fulfilled by** column in the request log.  The **Fulfilled by** column shows how a request was fulfilled.  If the request was fulfilled by your service worker or from the cache, that information is now tracked in the **Fulfilled by** column.  When exporting or importing the request log as a `HAR` file, the information in the **Fulfilled by** column is also included.
```

Reviewer 2: I felt like HAR needed a format. If "code" backticks are appropriate great. If not, no worries.

> In Microsoft Edge 100, the **Network** tool supports a new, **Fulfilled by** column in the request log.  The **Fulfilled by** column shows how a request was fulfilled.  If the request was fulfilled by your service worker or from the cache, that information is now tracked in the **Fulfilled by** column.  When exporting or importing the request log as a HAR file, the information in the **Fulfilled by** column is also included.

Reviewer: Is the comma after "new" grammatically correct? I also think this can be more concise. I think it reads better like this
```suggestion
In Microsoft Edge 100, the **Network** tool supports a new **Fulfilled by** column in the request log.  If the request was fulfilled by your service worker or from the cache, that information is now tracked in the **Fulfilled by** column.  When exporting or importing the request log as a HAR file, the information in the **Fulfilled by** column is also included.
```

![The Fulfilled By column in the request log of the Network tool.](devtools-100-images/fulfilled-by-request-log.png)
<!--
If you don't have the **Fulfilled by** column, right-click the table headers in the request log and make sure **Fulfilled by** is checked.
-->

To learn more about the request log in the Network tool, see [Filter requests by properties](../../../network/reference.md#display-a-log-of-requests) in _Network features reference_.  For more information about service workers, see [Service Worker improvements](../../../service-workers/index.md).

> For the history of this feature, see [Outlook: Add column in the Network activity log to track why resources are not loaded from DiskCache when ServiceWorker is present · Issue #16](https://github.com/MicrosoftEdge/DevTools/issues/16).

Reviewer: Again, I think this link text is too verbose. This seems much simpler:
```suggestion
For the history of this feature, see [Issue #16](https://github.com/MicrosoftEdge/DevTools/issues/16).
```

<!-- ====================================================================== -->
## When importing a performance profile, links now map to your original code

<!-- Title: Use sourcemaps from Azure Artifacts symbol server to better debug performance issues -->
<!-- Subtitle: Links from an imported performance profile now map to your original code because of source maps. -->

When recording a performance profile, the flame chart in the **Performance** tool displays links from each event on the main thread to the corresponding JavaScript function in the **Sources** tool.  However, these links are broken when you export and import a performance profile.

> In Microsoft Edge 100, the links from an imported performance profile now use source maps from Azure Artifacts symbol server to map back to your familiar, original source code.

Reviewer 2: Just wondering of the Azure Artifacts symbol server could be a link to the repo or the how-to article.

Author: Already linked to how-to article a few lines below, in like a See Also list.  
Design principles: 
Avoid multiple hyperlinks to same destination, confusing. 
Prefer standalone links over mid-sentence links.

> In Microsoft Edge 100, the links from an imported performance profile now use source maps from Azure Artifacts symbol server to map back to your familiar, original source code.

Reviewer 3: Suggest adding:

> The performance profile must have been exported in Microsoft Edge 100, which includes the information necessary to resolve source maps in the exported performance trace.

This is so people can understand that exporting from < 100 and importing into 100 won't allow it to magically work.

![Links from imported performance profile use source maps go to your familiar, original source code.](devtools-100-images/links-perf-profile-orig-source-code.png)

To learn more about using source maps in DevTools, see [Map the processed code to your original source code, for debugging](../../../javascript/source-maps.md).

> To get started with storing your source maps in Azure Artifacts symbol server and connecting to them from DevTools, see:

Reviewer 2: ```suggestion
To start storing your source maps in Azure Artifacts symbol server and connecting to them from DevTools, see:
```

*  [Securely debug original code by publishing source maps to the Azure Artifacts symbol server](../../../javascript/publish-source-maps-to-azure.md).
*  [Securely debug original code by using Azure Artifacts symbol server source maps](../../../javascript/consume-source-maps-from-azure.md).

<!-- ====================================================================== -->
## Fix: The Waterfall view in the Network tool now matches Visual Studio Code themes

<!-- Title: Themes from Visual Studio Code now apply to the Waterfall view -->
<!-- Subtitle: The Waterfall view of requests in the Network tool now match the VS Code themes. -->

In previous versions of Microsoft Edge, the **Waterfall** view of requests in the **Network** tool didn't match the themes from Visual Studio Code that apply to the rest of DevTools.  In Microsoft Edge 100, this issue has been fixed.

> The selected theme wasn't applied in the Waterfall view:

Reviewer: Can we add some additional information to this sentence before the image:
```suggestion
Before Microsoft Edge 100, the selected theme wasn't applied in the Waterfall view:
```

![The selected theme wasn't applied in the Waterfall view.](devtools-100-images/waterfall-view-requests-network-no-theme.png)

> The selected theme is now applied to the Waterfall view of requests in the Network tool:

Reviewer: Same thing as my previous comment. Can we add some additional information to this sentence before the image:
```suggestion
Now in Microsoft Edge 100, the selected theme is now applied to the Waterfall view of requests in the Network tool:
```
